### PR TITLE
Migrate build logic to Kotlin DSL

### LINKED ISSUE #186: Migrate build logic to Kotlin DSL
As Gradle went Kotlin DSL first this year, which should migrate all our plugins to Kotlin DSL and use it in the samples too.

This plugin seems to be relatively easily convertible.

### PR DESCRIPTION
**Pull request by 3flex**
Closes #186 

The diff from GitHub makes it hard to review, but the commit history should be clear.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    isRequired = providers.environmentVariable("CI").isPresent
+
+    useInMemoryPgpKeys(
+        providers.environmentVariable("PGP_SIGNING_KEY").orNull,
+        providers.environmentVariable("PGP_SIGNING_KEY_PASSPHRASE").orNull
+    )
+}
+
+githubRelease {
+    token(System.getenv("CCUD_GIT_TOKEN") ?: "")
```
Comment (Goooler) [inline build.gradle.kts:94]:
```suggestion
    token(providers.environmentVariable("CCUD_GIT_TOKEN").orNull.orEmpty())
```

**Review by Goooler — APPROVED** [review]

File: settings.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+plugins {
+    id("com.gradle.develocity") version "3.17"
+    id("com.gradle.common-custom-user-data-gradle-plugin") version "2"
```
Comment (clayburn) [inline settings.gradle.kts:3]:
```suggestion
    id("com.gradle.common-custom-user-data-gradle-plugin") version "2.0"
```

**Review by clayburn — CHANGES_REQUESTED** [review]
Thanks for the contribution @3flex! I made one suggestion. If you take that and fix the merge conflict, we'd be happy to accept this.

**Comment by 3flex** [comment]
Fixed - I've also updated README in the last commit as there were a few references (and a link) to settings.gradle. Please check that before merging, thanks!

**Review by clayburn — APPROVED** [review]

**Pull request merged**